src/components/logInModal/logInModal.js

```diff
@@ -0,0 +1,58 @@
+import { useState } from "react";
+import LogInForm from "../logInForm/LogInForm";
+import RegisterForm from "../registerForm/RegisterForm";
+import ForgotPasswordForm from "../forgotPasswordForm/ForgotPasswordForm";
+import AnonymousUserForm from "../anonymousUserForm/AnonymousUserForm";
+import styles from "@/styles/logInModal.module.scss";
+
+const LogInModal = ({ showLogInModal, handleShowLogInModal, formToShow }) => {
+  const [form, setForm] = useState(formToShow);
+  const setToRegister = () => {
+    setForm("register");
+  };
+  const setToForgotPassword = () => {
+    setForm("password");
+  };
+  const setToAnonymous = () => {
+    setForm("anonymous");
+  };
+  const setToLogin = () => {
+    setForm("login");
+  };
+
+  if (!showLogInModal) {
+    return null;
+  } else if (form === "login") {
+    return (
+      <div data-testid="logInForm" onClick={handleShowLogInModal} className={styles.logInModalContainer}>
+        <div onClick={(e)=>{e.stopPropagation();}} className={styles.logInModal}>
+          <LogInForm
+            setToRegister={setToRegister}
+            setToForgotPassword={setToForgotPassword}
+            setToAnonymous={setToAnonymous}
+          />
+        </div>
+      </div>
+    );
```

Comment:
This top-level div (with testid.. etc) is repeated throughout this component. You could rearrange the code to make it less repetitive. For instance, showing the data-testid inside the component, instead of in a wrapper, and perhaps using the logInModal class inside the respective Form-components.

I'm also wondering why the handleShowLogInModal is on the wrapper div here

Reply:
Good input! I will clean up the code and make it a bit less repetitive. 

Regarding the wrapper, this is how I solved the closing of the modal when it's open and you want to close it by clicking outside of it. The wrapper stretches across the whole screen and by clicking it the modal closes (also why I added the onClick event for the child div to stop it from triggering that event. I'm sure there is a better way to handle this, but I found it quite concise and easy.

Reply:
Gotcha. But you only have it on the login form, not the others. How would you close down the others? 

Mobile is also missing a way to close the modal, considering the size of it means it covers the entire screen.

Here's a suggestion for making the code more concise: 
```javascript
  let formElement;

  switch (form) {
   // I like the look of switch-case when dealing with strings like this, 
   // but obviously an if-else would work just as well 😁
    case "login":
      formElement = (
        <LogInForm
          setToRegister={setToRegister}
          setToForgotPassword={setToForgotPassword}
          setToAnonymous={setToAnonymous}
        />
      );
      break;
    case "register":
      formElement = (
        <RegisterForm setToLogin={setToLogin} setToAnonymous={setToAnonymous} />
      );
      break;
    case "password":
      formElement = <ForgotPasswordForm setToLogin={setToLogin} />;
      break;
    case "anonymous":
      formElement = <AnonymousUserForm setToLogin={setToLogin} />;
      break;
  }

  return (
   //  the following line is a very common way of 
   //  conditionally displaying components in React
    showLogInModal && (
      <div
   //  note: makes it loginForm, not logInForm, so one of the tests will fail
        data-testid={`${form}Form`}
        onClick={handleShowLogInModal}
        className={styles.logInModalContainer}
      >
        <div
          onClick={(e) => {
            e.stopPropagation();
          }}
          className={styles.logInModal}
        >
          {formElement}
        </div>
      </div>
    )
  );

Reply:
I'm not a huge fan of switch statments, but I cleaned up the code and made it a bit less repetitive.